Source/WebCore/page/NavigateEvent.h

```diff
@@ -81,22 +89,30 @@ class NavigateEvent final : public Event {
     String downloadRequest() { return m_downloadRequest; };
     JSC::JSValue info() { return m_info; };
 
-    void intercept(NavigationInterceptOptions&&);
-    void scroll();
+    ExceptionOr<void> intercept(NavigationInterceptOptions&&);
+    ExceptionOr<void> scroll();
+
+    bool wasIntercepted() const { return m_interceptionState != InterceptionState::None; };
 
 private:
     NavigateEvent(const AtomString& type, const Init&);
 
+    ExceptionOr<void> sharedChecks();
+
     NavigationNavigationType m_navigationType;
     RefPtr<NavigationDestination> m_destination;
     RefPtr<AbortSignal> m_signal;
     RefPtr<DOMFormData> m_formData;
     String m_downloadRequest;
+    Vector<RefPtr<NavigationInterceptHandler>> m_handlers;
     JSC::JSValue m_info;
     bool m_canIntercept { false };
     bool m_userInitiated { false };
     bool m_hashChange { false };
     bool m_hasUAVisualTransition { false };
+    InterceptionState m_interceptionState { InterceptionState::None };
```

Comment:
If InterceptionState is not specced, we could remove InterceptionState::None and instead make this a std::optional (to represent None).